### DISCUSSION TITLE
Missing event types in Angular?

### DISCUSSION THREAD

**Discussion by DerStimmler** [vidstack/player#1131] [Q&A]
Hello,

I'm using vidstack in an Angular application. I noticed that I can bind [all events mentioned in the docs](https://www.vidstack.io/docs/wc/player/components/core/player?styling=tailwind-css#player.events) for the `media-player` component in my template but I'm getting type errors for specific events.

For example the `playing` or `progress` events can be bound like this without a problem:
```
<media-player
    (progress)="log($event)"
   ...
  >
```

But for other events like `provider-change` I get the error
`Event provider-change is not emitted by any applicable directives nor by <media-player> element `.

```
<media-player
    (provider-change)="log($event)"
   ...
  >
```

Do I have to configure or install anything in addition to the docs to get proper type support?

I'm on vidstack `v1.10.3`, Angular `17.0.8`, typescript `5.2.2` and using WebStorm.

**Comment by mihar-22** [comment]
We don't have types for Angular at the moment, it's something I'll look into in the coming days. No idea at the moment how to generate them. If you're aware of how to add custom element types (props/events) to Angular please let me know :)

**Comment by naouaro** [comment]
> Hello,
> 
> I'm using vidstack in an Angular application. I noticed that I can bind [all events mentioned in the docs](https://www.vidstack.io/docs/wc/player/components/core/player?styling=tailwind-css#player.events) for the `media-player` component in my template but I'm getting type errors for specific events.
> 
> For example the `playing` or `progress` events can be bound like this without a problem:
> 
> ```
> <media-player
>     (progress)="log($event)"
>    ...
>   >
> ```
> 
> But for other events like `provider-change` I get the error `Event provider-change is not emitted by any applicable directives nor by <media-player> element `.
> 
> ```
> <media-player
>     (provider-change)="log($event)"
>    ...
>   >
> ```
> 
> Do I have to configure or install anything in addition to the docs to get proper type support?
> 
> I'm on vidstack `v1.10.3`, Angular `17.0.8`, typescript `5.2.2` and using WebStorm.

Hey there, been able to read all events using `addEventListener` no need for binding.

**Reply by DerStimmler** [reply]
Hi, yes with `addEventListener` I can read all events. I can even bind all events in the template properly. My IDE just shows an error that the event is not emitted by `media-player`.

**Comment by DerStimmler** [comment]
Ok I noticed that the error only occures on Rider and Webstorm. In VSCode, although I get no suggestions via intellisense, it shows no error and even type info.

![image](https://github.com/vidstack/player/assets/43130816/52105cb9-4e2f-40e3-b497-51359d5b860f)
